client/src/main/java/org/threadly/litesockets/client/http/HTTPClient.java

```diff
@@ -331,17 +331,27 @@ private void processQueue() {
   }
   
   private void process(HTTPRequestWrapper hrw) {
-    if(hrw != null) {
-      try {
-        hrw.updateReadTime();
-        hrw.client = getTCPClient(hrw.chr.getHTTPAddress());
-        inProcess.put(hrw.client, hrw);
-        hrw.client.write(hrw.chr.getHTTPRequest().getByteBuffer());
-        hrw.client.write(hrw.chr.getBodyBuffer().duplicate());
-      } catch (Exception e) {
-        //Have to catch all here or we dont keep processing if NoThreadSE is in use
-        //hrw.slf.setFailure(e);
+    if (hrw.slf.isDone()) {
+      // don't process, request either timed out while queued or otherwise completed in error
+      return;
+    }
+    try {
+      hrw.requestStarting();
+      hrw.client = getTCPClient(hrw.chr.getHTTPAddress());
+      inProcess.put(hrw.client, hrw);
+      SimpleMergedByteBuffers writeBuffer;
+      if (hrw.chr.getBodyBuffer() == null) {
+        writeBuffer = new SimpleMergedByteBuffers(false, 
+                                                  hrw.chr.getHTTPRequest().getByteBuffer());
+      } else {
+        writeBuffer = new SimpleMergedByteBuffers(false, 
+                                                  hrw.chr.getHTTPRequest().getByteBuffer(), 
+                                                  hrw.chr.getBodyBuffer().duplicate());
       }
+      hrw.client.write(writeBuffer);
+    } catch (Throwable t) {
+      //Have to catch all here or we dont keep processing if NoThreadSE is in use
+      hrw.slf.setFailure(t);
```

Comment:
This was previously commented out.  Uncommenting it did cause unit test failures due to the NPE I fixed above.  But it seems to me that this would make sense, otherwise we can't guarantee a request would finish.  Let me know if there is anything I am missing.

Reply:
Yeah, wish I could remember why it was commented out, could have just been a mistake, but as long as its working it makes sense.